fastlane/lib/fastlane/actions/set_build_number_repository.rb

```diff
@@ -13,7 +13,10 @@ def self.run(params)
           use_hg_revision_number: params[:use_hg_revision_number]
         )
 
-        Fastlane::Actions::IncrementBuildNumberAction.run(build_number: build_number)
+        Fastlane::Actions::IncrementBuildNumberAction.run(
+          build_number: build_number,
+          xcodeproj: params[:xcodeproj]
```

Comment:
This should be OK 👍  `xcodeproj` is an optional param to the other action, so it handles the possibility that the value is `nil`